test_tf2/CMakeLists.txt

```
@@ -9,7 +9,14 @@ if(NOT BUILD_TESTING)
   return()
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(NOT WIN32)
```

```suggestion
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

(see my comment in https://github.com/ros2/geometry2/pull/322#discussion_r488870267 for reasons why)